### PR TITLE
Scheduler - Horizontal virtual scrolling - Date table render

### PR DESCRIPTION
**Pull request by AntonSermyazhko**

### CODE REVIEW

File: js/renovation/ui/scheduler/workspaces/__tests__/utils.test.tsx [hunk 1]
```diff
@@ -65,6 +66,36 @@ describe('Workspaces utils', () => {
     });
   });
 
+  describe('addWidthToStyle', () => {
+    it('should return an empty obbject if height is undefined', () => {
```
Comment (AryamnovEugeniy) [inline js/renovation/ui/scheduler/workspaces/__tests__/utils.test.tsx]:
```suggestion
    it('should return an empty obbject if width is undefined', () => {
```

File: js/renovation/ui/scheduler/workspaces/__tests__/utils.test.tsx [hunk 2]
```diff
@@ -65,6 +66,36 @@ describe('Workspaces utils', () => {
...
+      expect(addWidthToStyle(undefined))
+        .toEqual({});
+    });
+
+    it('should return ucorrect style if height is provided', () => {
```
Comment (AryamnovEugeniy) [inline js/renovation/ui/scheduler/workspaces/__tests__/utils.test.tsx]:
```suggestion
    it('should return ucorrect style if width is provided', () => {
```

File: js/renovation/ui/scheduler/workspaces/__tests__/utils.test.tsx [hunk 3]
```diff
@@ -65,6 +66,36 @@ describe('Workspaces utils', () => {
...
+
+    it('should spread styles', () => {
+      expect(addWidthToStyle(500, { height: '300px' }))
+        .toEqual({
+          height: '300px',
+          width: '500px',
+        });
+    });
+
+    it('should spread styles when height is defined in restAttributes', () => {
```
Comment (AryamnovEugeniy) [inline js/renovation/ui/scheduler/workspaces/__tests__/utils.test.tsx]:
```suggestion
    it('should spread styles when width is defined in restAttributes', () => {
```

File: js/renovation/ui/scheduler/workspaces/base/__tests__/row.test.tsx [hunk 4]
```diff
@@ -35,5 +36,126 @@ describe('RowBase', () => {
...
+        widths: [50, 200],
+      },
+    }, {
+      hasLeftVirtualCell: true,
+      hasRightVirtualCell: true,
+      expected: {
+        selectors: [VirtualCell, '.child', VirtualCell],
+        widths: [100, 50, 200],
+      },
+    }].forEach((option) => {
```
Comment (AryamnovEugeniy) [inline js/renovation/ui/scheduler/workspaces/base/__tests__/row.test.tsx:67]: I would suggest to extract all of the necessary fields here

File: js/renovation/ui/scheduler/workspaces/base/__tests__/row.test.tsx [hunk 5]
```diff
@@ -35,5 +36,126 @@ describe('RowBase', () => {
...
+          hasLeftVirtualCell: true,
+        }, {
+          isVirtual: false,
+          leftVirtualCellWidth: 0,
+          hasLeftVirtualCell: false,
+        }, {
+          isVirtual: false,
+          leftVirtualCellWidth: 10,
+          hasLeftVirtualCell: false,
+        }].forEach((option) => {
```
Comment (AryamnovEugeniy) [inline js/renovation/ui/scheduler/workspaces/base/__tests__/row.test.tsx:107]: Same here - let us extract fields from an object

File: js/renovation/ui/scheduler/workspaces/base/date_table/__tests__/layout.test.tsx [hunk 6]
```diff
@@ -96,60 +100,111 @@ describe('DateTableLayoutBase', () => {
...
+          expectedBottom: 0,
+        }, {
+          top: 100,
+          bottom: 200,
+          expectedTop: 100,
+          expectedBottom: 200,
+        }].forEach(({
+          top, bottom, expectedTop, expectedBottom,
+        }) => {
+          it(`should been set correctly if viewData.topVirtualRowHeight=${top}, viewData.bottomVirtualRowHeight=${bottom}`, () => {
```
Comment (AryamnovEugeniy) [inline js/renovation/ui/scheduler/workspaces/base/date_table/__tests__/layout.test.tsx]:
```suggestion
          it(`should be calculated correctly if viewData.topVirtualRowHeight=${top}, viewData.bottomVirtualRowHeight=${bottom}`, () => {
```

File: js/renovation/ui/scheduler/workspaces/base/date_table/__tests__/layout.test.tsx [hunk 7]
```diff
@@ -96,60 +100,111 @@ describe('DateTableLayoutBase', () => {
...
+          expectedRight: 0,
+        }, {
+          left: 100,
+          right: 200,
+          expectedLeft: 100,
+          expectedRight: 200,
+        }].forEach(({
+          left, right, expectedLeft, expectedRight,
+        }) => {
+          it(`should been set correctly if viewData.leftVirtualCellWidth=${left}, viewData.rightVirtualCellWidth=${right}`, () => {
```
Comment (AryamnovEugeniy) [inline js/renovation/ui/scheduler/workspaces/base/date_table/__tests__/layout.test.tsx]:
```suggestion
          it(`should be calculated correctly if viewData.leftVirtualCellWidth=${left}, viewData.rightVirtualCellWidth=${right}`, () => {
```

File: js/renovation/ui/scheduler/workspaces/base/date_table/__tests__/layout.test.tsx [hunk 8]
```diff
@@ -96,60 +100,111 @@ describe('DateTableLayoutBase', () => {
...
+            expect(layout.rightVirtualCellWidth)
+              .toEqual(expectedRight);
+          });
+        });
+      });
 
-        expect(layout.virtualCellsCount)
-          .toBe(1);
+      describe('virtualCellsCount', () => {
+        it('should been set correctly', () => {
```
Comment (AryamnovEugeniy) [inline js/renovation/ui/scheduler/workspaces/base/date_table/__tests__/layout.test.tsx]:
```suggestion
        it('should be calculated correctly', () => {
```

**Review by AryamnovEugeniy — APPROVED** [review]

**Review by andrewmakarov — APPROVED** [review]

**Pull request merged**